streams/lms_metadata.py

```diff
@@ -0,0 +1,145 @@
+  def __init__(self, name: str, meta_ref: Optional[int] = 2, dump: Optional[bool] = False):
+    self.player_name = name
+    self.IP = None
+    self.meta_ref_rate = meta_ref
+    self.connected = False
+    self.dump = dump
+
+  def connect(self):
+    """Discovers LMS Player and then requests metadata repetitively"""
+    f = open(f"lms_{str(self.player_name).replace(' ', '_')}_metadata.json", 'wt', encoding='utf-8')
+    json.dump({'track': 'Loading...', 'artist': 'Loading...', 'album': 'Loading...', 'image_url': 'static/imgs/lms.png'}, f, indent = 2)
+    f.close()
+    x = 0
+
+    # Much faster method of connecting to the metadata server using code from: https://github.com/ralph-irving/squeezelite/blob/master/tools/find_server.c
+    IPFind = subprocess.run('streams/find_server', capture_output=True, text=True)
+    # Uses re.search because find_server.c spits out as '{Hostname}:{port} ({IP})', so I scrape the data from inbetween the parentheses to get the IP
+    self.IP = re.search(r"\((.*?)\)", IPFind.stdout).group(1)
+
+
+    # When not connected, search for player to connect to by the proper name
+    while not self.connected:
+      try:
+        player_json = {"id": 1,	"method": "slim.request",	"params": [self.player_name, ["players", "-", 100, "playerid"]]}
+        player_info = requests.get(f'http://{self.IP}:9000/jsonrpc.js', json=player_json, timeout=10)
+        player_load = json.loads(player_info.text)
+        players = player_load['result']['players_loop']
+
+        for player in players:
+          connected = player['connected']
+          if connected and player['name'] == self.player_name:
+            self.connected = True
+          else:
+            time.sleep(0.1)
+      except:
+        # When first creating an LMS stream, there can be random errors that will close the while loop
+        # typically when asking the player for info when there isn't a player linked to the stream yet
+        print("FAIL")
+        pass
+
+    while self.connected:
+      try:
+        track_json = {"id": 1, "method": "slim.request", "params": [ self.player_name, ["status", "-",100] ]}
+        track_info = requests.post(f'http://{self.IP}:9000/jsonrpc.js 2>/dev/null', json=track_json, timeout=200)
+        track_load = json.loads(track_info.text)
+
+        track_id = track_load["result"]["playlist_loop"][0]["id"]
+        song_json = {"id":2,"method":"slim.request","params":[ self.player_name, ["songinfo","-",100,f"track_id:{track_id}"]]}
+        song_info = requests.post(f'http://{self.IP}:9000/jsonrpc.js 2>/dev/null', json=song_json, timeout=200)
+        song_load = json.loads(song_info.text)
+
+        x = 0
+        song_data = {}
+        for item in song_load['result']['songinfo_loop']:
+          for info in item:
+            song_data[info] = song_load['result']['songinfo_loop'][x][info]
+          x += 1
```

Comment:
this could be replaced with something like:
```python
  def flatten(lms_info: dict) -> dict:
    flat_data = {}
    for item in lms_info:
      for info in item:
        flat_data[info] = lms_info[x][info]
      x += 1
    return flat_data
```